### ISSUE TITLE
lvl=warn msg="Database compacting, degraded performance" and stopped syncing

### ISSUE THREAD

**Issue by shiziwen** [bnb-chain/bsc#1687]
Hi, I have two bsc fullnode, and I got lots of msg:
```
t=2023-06-09T16:18:24+0000 lvl=warn msg="Database compacting, degraded performance" database=/data/bsc/node/geth/chaindata
t=2023-06-09T16:19:24+0000 lvl=warn msg="Database compacting, degraded performance" database=/data/bsc/node/geth/chaindata
t=2023-06-09T16:20:27+0000 lvl=warn msg="Database compacting, degraded performance" database=/data/bsc/node/geth/chaindata
t=2023-06-09T16:21:28+0000 lvl=warn msg="Database compacting, degraded performance" database=/data/bsc/node/geth/chaindata
t=2023-06-09T16:22:31+0000 lvl=warn msg="Database compacting, degraded performance" database=/data/bsc/node/geth/chaindata
t=2023-06-09T16:23:31+0000 lvl=warn msg="Database compacting, degraded performance" database=/data/bsc/node/geth/chaindata
t=2023-06-09T16:24:31+0000 lvl=warn msg="Database compacting, degraded performance" database=/data/bsc/node/geth/chaindata
t=2023-06-09T16:25:31+0000 lvl=warn msg="Database compacting, degraded performance" database=/data/bsc/node/geth/chaindata
```

and the node stopped syncing.

The version is :
```
./geth_linux version
Geth
Version: 1.2.4
Git Commit: 3c5f54fc805994b0b60b50dbd59e4424df25f17a
Git Commit Date: 20230523
Architecture: amd64
Go Version: go1.19.9
Operating System: linux
GOPATH=
GOROOT=/opt/hostedtoolcache/go/1.19.9/x64
```

How can I fixed it?

I got two times of this issue today!

**Comment by deepcrazy** [comment]
Hi there,

I think we have already discussed regarding this in discord and have suggested you to upgrade to the latest version v1.2.5: https://github.com/bnb-chain/bsc/releases/tag/v1.2.5.
Also, This `Database compacting` should not stay for long time. It should go away automatically. If not, then I would suggest to restart your node once by adding `maxpeers` flag and set to 200 once and then try with 500 on next try and let it run for a while and check if it still works.

Thanks.

**Comment by J1a-wei** [comment]
Hi team @deepcrazy ,  I have encountered the same issue, but the version（v1.2.4) I am using is slightly outdated

 What's the motivation to trigger this situation. Is  it bug ? Does this happen automatically? This is very unfriendly for operators, because it suddenly stops synchronizing blocks. I searched some documents. It seems that there is a parameter that can turn off data compression(--nocompaction), but will this cause other problems?
